### PR TITLE
Updated release notes and README to prepare for 3.0a3

### PR DESCRIPTION
**Pull request by modelonrobinandersson**

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -4,6 +4,21 @@ The release notes are typically a highlighting subset of all changes made. For f
 
 Note that version 2.1 is the first version with release notes. Please see the commit history for older versions.
 
+## 3.0a3
+
+### Improvements
+
+- Improved error handling.
+- Attribute buffers are now properly cleared if an elements fail to parse.
```
Comment (filip-stenstrom) [inline RELEASE_NOTES.md:12]:
```suggestion
- Attribute buffers are now properly cleared if an element fails to parse.
```

File: RELEASE_NOTES.md [hunk 2]
```diff
@@ -4,6 +4,21 @@ The release notes are typically a highlighting subset of all changes made. For f
...
+- Disabled jm_locale_test if locale testing is disabled.
+### Changes
```
Comment (filip-stenstrom) [inline RELEASE_NOTES.md:14]:
```suggestion
- Disabled jm_locale_test if locale testing is disabled.

### Changes
```

File: RELEASE_NOTES.md [hunk 3]
```diff
@@ -4,6 +4,21 @@ The release notes are typically a highlighting subset of all changes made. For f
...
+
+- Resolved reported vulnerability issues by:
+    - Updated to expat 2.5.0 up from version 2.8.0.
+    - Updated to zlib 1.2.13 up from version 1.2.6.
```
Comment (filip-stenstrom) [inline RELEASE_NOTES.md:18]:
```suggestion
- Resolved reported vulnerability issues:
    - Updated to expat 2.5.0 (from version 2.8.0).
    - Updated to zlib 1.2.13 (from version 1.2.6).
```

File: RELEASE_NOTES.md [hunk 4]
```diff
@@ -4,6 +4,21 @@ The release notes are typically a highlighting subset of all changes made. For f
...
+- Renamed `default_experiment_step` to `_step_size`.
```
Comment (filip-stenstrom) [inline RELEASE_NOTES.md:19]: This doesn't make any sense to me.

**Review by PeterMeisrimelModelon — APPROVED** [review]

File: RELEASE_NOTES.md (hunk 3 above)
Comment (ni-nutkalit) [reply RELEASE_NOTES.md]:
```
- Updated to expat 2.5.0 (from version 2.4.8). ?
```

Comment (modelonrobinandersson) [reply RELEASE_NOTES.md]: Good catch, thank you!

**Pull request merged**